docs/src/views/Theming.tsx

```diff
@@ -121,42 +120,92 @@ export default () => (
+      render={() => <Text styles={{ color: 'green' }}>This is green text</Text>}
+    />
+    <p>
+      Every slot (named part) of every component also accepts <code>styles</code> that are applied
+      to the root element of the slot.
     </p>
-
     <ExampleSnippet
       value={[
         `<Button`,
-        `  icon={{ name: "user", styles: { borderBottom: '4px solid red' } }}`,
+        `  icon={{ name: 'user', styles: { boxShadow: '0 0 0 2px red' } }}`,
         `  content="Profile"`,
         `/>`,
       ].join('\n')}
       render={() => (
-        <Button
-          icon={{ name: 'user', styles: { borderBottom: '4px solid red' } }}
-          content="Profile"
-        />
+        <Button icon={{ name: 'user', styles: { boxShadow: '0 0 0 2px red' } }} content="Profile" />
       )}
     />
 
     <p>
-      Also you can style the <code>color</code> css property of a <code>Text</code> component.
+      You can also define <code>styles</code> for all components in a part of your render tree using
+      the <NavLink to="components/provider">Provider</NavLink>.
+    </p>
+    <p>
+      This is done with the Provider's <code>theme</code> prop. Styles are applied based on
+      component display name and slot name. Here's how we can style the <code>Button</code> and its{' '}
+      <code>icon</code> slot.
     </p>
     <ExampleSnippet
-      label="js"
       value={[
-        `const style = { color: 'green' }`,
+        `<Button icon="user" content="Profile" />`,
+        `<Button icon="user" content="Profile" />`,
         ``,
-        `<Text styles={{ style }}>This is green text</Text>`,
+        `<Provider`,
+        `  theme={{`,
+        `    componentStyles: {`,
+        `      Button: {`,
+        `        root: { boxShadow: '0 0 0 2px blue' },`,
+        `        icon: { boxShadow: '0 0 0 2px red' },`,
+        `        content: { boxShadow: '0 0 0 2px green' },`,
+        `      },`,
+        `    },`,
+        `  }}`,
+        `>`,
+        `  <span>`,
+        `    <Button icon="user" content="Profile" />`,
+        `    <Button icon="user" content="Profile" />`,
+        `  </span>`,
+        `</Provider>`,
       ].join('\n')}
-      render={() => <Text styles={{ color: 'green' }}>This is green text.</Text>}
+      render={() => (
+        <div>
+          <Button icon="user" content="Profile" />
+          <Button icon="user" content="Profile" />
+
+          <Provider
+            theme={{
+              componentStyles: {
+                Button: {
+                  root: { boxShadow: '0 0 0 2px blue' },
+                  icon: { boxShadow: '0 0 0 2px red' },
+                  content: { boxShadow: '0 0 0 2px green' },
+                },
+              },
+            }}
+          >
+            <span>
+              <Button icon="user" content="Profile" />
+              <Button icon="user" content="Profile" />
+            </span>
+          </Provider>
+        </div>
+      )}
     />
+    <p>
```

Comment:
just as a side notice - currently there are no links to the next (if necessary, as there are components sections coming next) and (surely) previous docs page - the ones that are present for the rest of the docs pages. Most probably you already have a plan to address that